mysensors/mysensors.py

```diff
@@ -502,9 +501,33 @@ def __init__(self, host, event_callback=None,
         self.sock = None
         self.server_address = (host, port)
         self.timeout = timeout
+        self.tcp_check_timer = time.time()
+        self.tcp_disconnect_timer = time.time()
         self.reconnect_timeout = reconnect_timeout
         self._stop_event = threading.Event()
 
+    def _check_connection(self):
+        """Check if connection is alive every reconnect_timeout seconds."""
+        if ((self.tcp_disconnect_timer + 2 * self.reconnect_timeout) <
+                time.time()):
+            self.tcp_disconnect_timer = time.time()
+            self.disconnect()
+            _LOGGER.info('No response. Disconnected.')
+            return
+        if not (self.tcp_check_timer + self.reconnect_timeout) < time.time():
+            return
+        msg = Message().copy(
+            child_id=255, type=self.const.MessageType.internal,
+            sub_type=self.const.Internal.I_VERSION)
+        self.fill_queue(msg.encode)
+        self.tcp_check_timer = time.time()
+
+    def _handle_internal(self, msg):
+        if msg.sub_type == self.const.Internal.I_VERSION:
+            self.tcp_disconnect_timer = time.time()
+        else:
+            super()._handle_internal(msg)
```

Comment:
You have to return the result of this call to the logic method which makes the call to the method here.
```py
return super()._handle_internal(msg)
```